1_7_R4/src/main/java/net/wesjd/anvilgui/version/Wrapper1_7_R4.java

```diff
@@ -0,0 +1,123 @@
+package net.wesjd.anvilgui.version;
+
+import net.minecraft.server.v1_7_R4.*;
+import org.bukkit.craftbukkit.v1_7_R4.entity.CraftPlayer;
+import org.bukkit.craftbukkit.v1_7_R4.event.CraftEventFactory;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.Inventory;
+
+/**
+ * {@link VersionWrapper} implemented for NMS version 1_7_R4
+ *
+ * @author Hasan Demirtaş
+ * @since 1.3.0
+ */
+public class Wrapper1_7_R4 implements VersionWrapper {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getNextContainerId(final Player player, final Object container) {
+        return this.toNMS(player).nextContainerCounter();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleInventoryCloseEvent(final Player player) {
+        CraftEventFactory.handleInventoryCloseEvent(this.toNMS(player));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void sendPacketOpenWindow(final Player player, final int containerId, final String guiTitle) {
+        this.toNMS(player).playerConnection.sendPacket(new PacketPlayOutOpenWindow(containerId, 8, "Repairing", 9, true));
```

Comment:
This case is from mode coder pack(mcp) sources. I tested and It's working on 1.7.10.